Sources/Data Model/HoldoutConfig.swift

```diff
@@ -73,25 +83,41 @@ struct HoldoutConfig {
         }
     }
     
+    /// Returns the applicable holdouts for the given flag ID by combining global holdouts (excluding any specified) and included holdouts, in that order.
+    /// Caches the result for future calls.
+    /// - Parameter id: The flag identifier.
+    /// - Returns: An array of `Holdout` objects relevant to the given flag.
     mutating func getHoldoutForFlag(id: String) -> [Holdout] {
         guard !allHoldouts.isEmpty else { return [] }
         
+        // Check cache and return persistent holdouts
         if let holdouts = flagHoldoutsMap[id] {
             return holdouts
         }
         
-        if let included = includedHoldouts[id], !included.isEmpty {
-            flagHoldoutsMap[id] = global + included
-        } else {
-            let excluded = excludedHoldouts[id] ?? []
-            let filteredHoldouts = others.filter { holdout in
+        var activeHoldouts: [Holdout] = []
+        
+        let excluded = excludedHoldouts[id] ?? []
+        
+        if !excluded.isEmpty {
+            activeHoldouts = global.filter { holdout in
                 return !excluded.contains(holdout)
             }
-            flagHoldoutsMap[id] = global + filteredHoldouts
+        } else {
+            activeHoldouts = global
         }
+        
+        let includedHoldouts = includedHoldouts[id] ?? []
+        
+        activeHoldouts += includedHoldouts
+        
+        flagHoldoutsMap[id] = activeHoldouts
+        cacheWriteCount += 1
```

Comment:
Can we remove this cache count - I understand this is for testing, but should not be copied to other SDKs.
We can add unit test to validate cache is updated and used.